src/producer/browser/Full.js

```diff
@@ -29,30 +29,66 @@ const FullBrowserProducer = (context) => {
   const splitsUpdater = SplitChangesUpdater(context);
   const segmentsUpdater = MySegmentsUpdater(context);
 
+  let isSynchronizeSplitsRunning = false;
```

Comment:
these vars are weird.. don't base them on the name of the method you use today but rather the action of synchronizing Splits. This is a boolean flag that indicates if the synchronization of splits is running, is doesn't actually mean the method below (synchronizeSplits) is currently being executed.

isSynchronyzingSplits and isSynchronizingSegments ?

Reply:
changed to `isSynchronyzingSplits`, `isSynchronizing[My]Segments`